nsqd/nsqd.go

```diff
@@ -203,13 +206,14 @@ func (n *NSQD) getFlag(f int32) bool {
 }
 
 func (n *NSQD) SetHealth(err error) {
-	n.errMtx.Lock()
-	defer n.errMtx.Unlock()
-	n.err = err
 	if err != nil {
 		n.setFlag(flagHealthy, false)
```

Comment:
I think we need `flagHealthy` internally to side step the allocation for the normal case of `SetHealth(nil)`.

Gut feeling is the flag check is [probably faster](https://golang.org/src/sync/atomic/value.go?s=732:770#L17).